camel/models/phi_model.py

```diff
@@ -0,0 +1,150 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+import base64
+import io
+from typing import Any, Dict, List, Optional, Union
+
+from openai import Stream
+from PIL import Image
+from vllm import LLM, SamplingParams
+
+from camel.configs import PHI_API_PARAMS, PHIConfig
+from camel.messages import OpenAIMessage
+from camel.models import BaseModelBackend
+from camel.types import (
+    ChatCompletion,
+    ChatCompletionChunk,
+    CompletionUsage,
+    ModelType,
+)
+from camel.utils import BaseTokenCounter, OpenAITokenCounter
+
+
+class PHIModel(BaseModelBackend):
+    def __init__(
+        self,
+        model_type: ModelType,
+        model_config_dict: Dict[str, Any],
+        api_key: Optional[str] = None,
+        url: Optional[str] = None,
+        token_counter: Optional[BaseTokenCounter] = None,
+    ) -> None:
+        super().__init__(
+            model_type=model_type,
+            model_config_dict=model_config_dict,
+            api_key=api_key,
+            url=url,
+        )
+        self._token_counter = token_counter
+        self.config = PHIConfig(**model_config_dict)
+        self.llm = LLM(
+            model=self.config.model,
+            trust_remote_code=self.config.trust_remote_code,
+            max_model_len=self.config.max_model_len,
+            limit_mm_per_prompt=self.config.limit_mm_per_prompt,
+        )
+
+    @property
+    def token_counter(self) -> BaseTokenCounter:
+        if not self._token_counter:
+            self._token_counter = OpenAITokenCounter(ModelType.GPT_4O_MINI)
+        return self._token_counter
+
+    def run(
+        self,
+        messages: List[OpenAIMessage],
+    ) -> Union[ChatCompletion, Stream[ChatCompletionChunk]]:
+        question = messages[1]['content'][0]['text']  # type: ignore[index]
+        image_urls = messages[1]['content'][1]['image_url']  # type: ignore[index]
```

Comment:
what if there's no `image_url ` in the message? I think user could also use Phi do some pure text-based task